src/SUFST/src/can_unpack.c

```diff
@@ -59,16 +89,22 @@ void queue_receive_thread_entry(ULONG input)
     {
         int ret;
         
-
         /* Receive data from the queue. */
-        ret = tx_queue_receive(unpack_ptr->rx_queue, &rx_msg, TX_WAIT_FOREVER);
+        ret = tx_queue_receive(&unpack_ptr->rx_queue,
+                                    (void *) &rx_msg_ptr,
+                                    TX_WAIT_FOREVER);
         if (ret != TX_SUCCESS)
         {
             return;
         }
+        memcpy((void*) rx_msg.data, (void*) rx_msg_ptr->data, rx_msg_ptr->length);
+
+        // mark the original received message as consumed
+        rtcan_msg_consumed(&unpack_ptr->rtcan, rx_msg_ptr);
```

Comment:
Why is the message copied locally and freed? I can't see any reason why this is necessary. The whole point of RTCAN distributing pointers to messages is so that there is no unnecessary copying. It should be possible to just pass `rx_msg_ptr->data` to the unpacking function, then free the RTCAN message once the unpacking has been done (also remember to free the message if any errors occur).